.github/workflows/ci-cd.yml

```
@@ -124,9 +124,24 @@ jobs:
         - ubuntu
         - windows
         - macos
+        tag:
+        - ''
+        - 'musllinux'
+        exclude:
+        - os: windows
+          tag: 'musllinux'
+        - os: macos
+          tag: 'musllinux'
+        - os: ubuntu
+          tag: >-
+           ${{
+           (github.event_name != 'push' || !contains(github.ref, 'refs/tags/'))
```

There's a more elegant shortcut in GHA nowadays:
```suggestion
           (github.event_name != 'push' || github.ref_type != 'tag')
```